frontend/admin/src/js/components/user/UserTableFilterDialog.tsx

```diff
@@ -0,0 +1,284 @@
+            </Item>
+          )}
+          <Item>
+            <SelectFieldV2
+              forceArrayFormValue
+              id="employmentDuration"
+              label={formatMessage({
+                defaultMessage: "Duration Preferences",
+              })}
+              options={optionsData.employmentDuration}
+            />
+          </Item>
+          <Item grow>
+            <MultiSelectFieldV2
+              id="jobLookingStatus"
+              label={formatMessage({
+                defaultMessage: "Availability",
+              })}
+              options={optionsData.jobLookingStatus}
+            />
+          </Item>
+          <Item>
+            <SelectFieldV2
+              forceArrayFormValue
+              id="profileComplete"
+              label={formatMessage({
+                defaultMessage: "Profile Complete",
+              })}
+              options={optionsData.profileComplete}
+            />
+          </Item>
+        </Row>
+        <Row>
+          <Item grow>
+            <MultiSelectFieldV2
+              id="skills"
+              label={formatMessage({
+                defaultMessage: "Skill Filter",
+              })}
+              options={optionsData.skills}
+              isLoading={rawGraphqlResults.skills.fetching}
+            />
+          </Item>
+          <Item>
+            <SelectFieldV2
+              forceArrayFormValue
+              id="govEmployee"
+              label={formatMessage({
+                defaultMessage: "Government Employee",
+              })}
+              options={optionsData.govEmployee}
+            />
+          </Item>
+        </Row>
+        <Dialog.Footer>
+          <Footer />
+        </Dialog.Footer>
+      </BasicForm>
+    </Dialog>
+  );
+};
+
+type UserTableFilterButtonProps = Pick<
+  UserTableFilterDialogProps,
+  "onSubmit" | "enableEducationType"
+> & { isOpenDefault?: boolean };
+const UserTableFilterButton = ({
+  onSubmit,
+  isOpenDefault = false,
+  ...rest
+}: UserTableFilterButtonProps) => {
+  const { emptyFormValues } = useFilterOptions();
+  const [activeFilters, setActiveFilters] =
+    useState<FormValues>(emptyFormValues);
+  const [isOpen, setIsOpen] = useState(isOpenDefault);
+
+  const handleOpen = () => setIsOpen(true);
+  const handleDismiss = () => setIsOpen(false);
+  const handleSubmit: SubmitHandler<FormValues> = (data) => {
+    onSubmit(data);
+    setActiveFilters(data);
+    setIsOpen(false);
+  };
+
+  return (
+    <>
+      <Button
+        onClick={handleOpen}
+        mode="outline"
+        color="black"
+        type="button"
+        data-h2-display="b(inline-flex)"
+        data-h2-align-items="b(center)"
+      >
+        <ButtonIcon icon={FilterIcon} />
+        <span>
+          <FormattedMessage
+            defaultMessage="Filter"
+            description="Text label for button to open filter dialog on admin tables."
+          />
+        </span>
+      </Button>
+      <UserTableFilterDialog
```

Comment:
So... you're going to hate me.... but there is so much going on this dialog that I think I'd like a nice fat test suite in jest.  Basically, confirming that if controls are interacted with, the emitted filter object is accurate.  I could be convinced that that should be it's own issue since it would be quite a bit of effort.  I think it's reasonable to at least a add a few jest tests for the new Multiselect though.